ui/src/message_popup/experience_page.jsx

```diff
@@ -91,56 +64,69 @@ export default React.createClass({
   },
   
   addResponseTime(time){
-    this.setState({responseTimes: this.state.responseTimes.concat(time)});
+    var gameSession = {...this.state.gameSession};
+    gameSession.responseTimes.push(time);
+    this.setState({ gameSession });
   },
 
   onLog(type, response:Response) {
     Api.logEvidence(type, {
       ...response,
-      name: this.state.email,
-      sessionId: this.state.sessionId,
+      name: this.state.gameSession.email,
+      sessionId: this.state.gameSession.sessionId,
       clientTimestampMs: new Date().getTime()
     });
   },
   
   onQuestionDone(elapsedSeconds) {
     this.playToast();
     this.addResponseTime(elapsedSeconds);
-    this.setState({ questionsAnswered: this.state.questionsAnswered + 1 });
+    var gameSession = {...this.state.gameSession};
+    gameSession.questionsAnswered += 1;
+    this.setState({ gameSession });
   },
 
   onDonePressed() {
     Routes.navigate(Routes.Home);
   },
   
   resetExperience(){
-    this.setState(this.getInitialState());
+    this.replaceState(this.getInitialState());
```

Comment:
I'm assuming this is to remove `gameSession`?  It's good to explicitly set that to null anyway, and then this will work as the regular `setState` as well.